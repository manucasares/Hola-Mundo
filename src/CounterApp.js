import React, { useState } from "react";

const CounterApp = ({value = 10}) => {


    const [count, setCount] = useState(value);

    //FUNCIONES
    const incrementar = () => {
        setCount(prevCount => prevCount + 1);
    };

    const decrementar = () => {
        setCount(prevCount => prevCount - 1);
    };

    const reset = () => {
        setCount(value);
    };

    return (
        <>
            <button onClick={decrementar}>-</button>
            <h2>{count}</h2>
            <button onClick={incrementar}>+</button>
            <button onClick={reset}>RESET</button>
        </>
    );
};

export default CounterApp;
