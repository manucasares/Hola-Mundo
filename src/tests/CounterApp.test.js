import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("Pruebas en Counterapp ", () => {

    let wrapper = shallow(<CounterApp />);

    beforeEach(()=>{
        wrapper = shallow(<CounterApp />);
    });


    test("debe de mostrar CounterApp correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de mostrar el valor por defecto de 100", () => {
        const value = "100";
        const wrapper = shallow(<CounterApp value={value} />);
        const expected = wrapper.find("h2").text();
        expect(expected).toBe(value);
    });

    test('el boton sum debe incrementar +1', ()=>{
        wrapper.find('button').at(1).simulate('click');
        const expected = wrapper.find('h2').text();
        expect(expected).toBe('11');
    });

    test('el boton sum debe decrementar -1', ()=>{
        wrapper.find('button').at(0).simulate('click');
        const expected = wrapper.find('h2').text();
        expect(expected).toBe('9');
    });

});
