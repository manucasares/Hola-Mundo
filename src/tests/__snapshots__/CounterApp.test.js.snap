// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Counterapp  debe de mostrar CounterApp correctamente 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    RESET
  </button>
</Fragment>
`;

exports[`Pruebas en Counterapp  debe de mostrar CounterApp correctamente 2`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    RESET
  </button>
</Fragment>
`;

exports[`Pruebas en Counterapp  debe de mostrar el valor por defecto de 100 1`] = `[Function]`;
